python/common/org/Python.java

```diff
@@ -915,10 +915,60 @@ public static void initializeModule(java.lang.Class cls, java.util.Map<java.lang
             "With a single iterable argument, return its biggest item. The\n" +
             "default keyword-only argument specifies an object to return if\n" +
             "the provided iterable is empty.\n" +
-            "With two or more arguments, return the largest argument.\n"
+            "With two or more arguments, return the largest argument.\n",
+        args = {"first"},
+        varargs = "others",
+        kwargs = "kwargs"
     )
-    public static org.python.Object max() {
-        throw new org.python.exceptions.NotImplementedError("Builtin function 'max' not implemented");
+    public static org.python.Object max(org.python.Object first, org.python.types.Tuple rest, org.python.types.Dict kwargs) {
+        org.python.Object val = null;
+        if (first == null) {
+            throw new org.python.exceptions.TypeError("max expected 1 arguments, got 0");
+        } else {
+            org.python.Iterable iter;
+            if (first instanceof org.python.types.List) {
```

Comment:
You shouldn't do type-checking, the max builtin function should work on any iterable object (namely, any object that has an `__iter__` method that returns an iterator), not just on lists/tuples/dicts/etc.

Example:

```
>>> class Deck:
...     cards = ['A'] + [str(i) for i in range(1, 10)] + ['J', 'Q', 'K']
...     def __iter__(self):
...         return iter(self.cards)
... 
>>> max(Deck())
'Q'
```

[Look here for an example of iterating over an iterable](https://github.com/pybee/voc/blob/master/python/common/org/Python.java#L1395-L1403).